src/Storefront/Resources/app/storefront/src/plugin/main-menu/offcanvas-menu.plugin.js

```diff
@@ -74,9 +80,28 @@ export default class OffcanvasMenuPlugin extends Plugin {
         OffCanvas.open(this._content, this._registerEvents.bind(this), this.options.position);
         OffCanvas.setAdditionalClassName(this.options.additionalOffcanvasClass);
 
+        this._removeNotNeededBackdrops();
+
         this.$emitter.publish('openMenu');
     }
 
+    _openMenuViaUrlParameter() {
+        const urlParams = new URLSearchParams(window.location.search);
+        if (urlParams.has('offcanvas') && urlParams.get('offcanvas') === 'menu') {
+            document.querySelector('[data-off-canvas-menu="true"]')?.click();
+        }
+    }
+
+    _removeNotNeededBackdrops() {
```

Comment:
The reason for the duplicate backdrops is, that your code is running twice. There are two `data-off-canvas-menu="true"` instances. And both run into their registerEvents and click the `data-off-canvas-menu="true"` again. And then open both at the same time.

I find it a bit questionable 🤔 to have two individual `data-off-canvas-menu="true"` in the template for mobile and tablet while they are rendered in an almost identical place.

![image](https://github.com/user-attachments/assets/11956a65-52d4-4973-95ae-8493358910e6)